pkg/build/graph/analysis/bc.go

```diff
@@ -238,6 +238,22 @@ func FindPendingTags(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
 	return markers
 }
 
+// getImageStreamTagSuggestion will return the appropriate marker Suggestion for when a BuildConfig is missing its input ImageStreamTag;  in particular,
+// it will determine whether or not another BuildConfig can produce the aforementioned ImageStreamTag
+func getImageStreamTagSuggestion(g osgraph.Graph, f osgraph.Namer, tagNode *imagegraph.ImageStreamTagNode, bcNode *buildgraph.BuildConfigNode) osgraph.Suggestion {
+	bcs := []string{}
+	for _, bcNode := range g.PredecessorNodesByEdgeKind(tagNode, buildedges.BuildOutputEdgeKind) {
+		bcs = append(bcs, f.ResourceName(bcNode))
+	}
+	if len(bcs) == 1 {
+		return osgraph.Suggestion(fmt.Sprintf("%s needs to be imported or created by executing `oc start-build %s`.", f.ResourceName(tagNode), strings.Join(bcs[:], ",")))
```

Comment:
On Mon, Apr 11, 2016 at 10:13 AM, Michail Kargakis <notifications@github.com
> wrote:

> In pkg/build/graph/analysis/bc.go
> <https://github.com/openshift/origin/pull/8351#discussion_r59210411>:
>
> > @@ -238,6 +238,22 @@ func FindPendingTags(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
> >  	return markers
> >  }
> >
> > +// getImageStreamTagSuggestion will return the appropriate marker Suggestion for when a BuildConfig is missing its input ImageStreamTag;  in particular,
> > +// it will determine whether or not another BuildConfig can produce the aforementioned ImageStreamTag
> > +func getImageStreamTagSuggestion(g osgraph.Graph, f osgraph.Namer, tagNode *imagegraph.ImageStreamTagNode, bcNode *buildgraph.BuildConfigNode) osgraph.Suggestion {
> > +	bcs := []string{}
> > +	for _, bcNode := range g.PredecessorNodesByEdgeKind(tagNode, buildedges.BuildOutputEdgeKind) {
> > +		bcs = append(bcs, f.ResourceName(bcNode))
> > +	}
> > +	if len(bcs) == 1 {
> > +		return osgraph.Suggestion(fmt.Sprintf("%s needs to be imported or created by executing `oc start-build %s`.", f.ResourceName(tagNode), strings.Join(bcs[:], ",")))
>
> suggest just the command
>

pushed

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8351/files/04c49b3cc4dea4c33a66eb33ac3118bf458329ee#r59210411>
>